File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
  let max = Math.max(...nums)+1;
    let smaller = new Array(max).fill(0);
    let sum = 0;
    for(let i = 0; i<nums.length;i++){
        smaller[nums[i]]++;
    }
    for(let i = 1 ; i < smaller.length ; i++){
        smaller[i] += smaller[i-1]
       
    }
    for(let i = 0 ; i<nums.length ; i++){
        let position = nums[i];
        if(position === 0) nums[i] = 0;
        else{
            nums[i] = smaller[position-1];
            
        }
    }
    return nums;
};

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:    
        count = 0
        result = []
        for i in range(0, len(nums)):
            for j in range(0, len(nums)):
                if nums[i] > nums[j]:
                    count += 1;
            result.append(count)
            count = 0
        return result
                
        